internal/server/iigointernal/judiciary.go

```diff
@@ -81,11 +81,31 @@ func (j *judiciary) PayPresident() shared.Resources {
 	return hold
 }
 
-// InspectHistory checks all actions that happened in the last turn and audits them.
+// inspectHistory checks all actions that happened in the last turn and audits them.
 // This can be overridden by clients.
 func (j *judiciary) inspectHistory(iigoHistory []shared.Accountability) (map[shared.ClientID]roles.EvaluationReturn, bool) {
 	j.budget -= serviceCharge
-	return j.clientJudge.InspectHistory(iigoHistory)
+	finalResults := getBaseEvalResults(shared.TeamIDs)
+	if j.clientJudge.HistoricalRetributionEnabled() {
+		for _, v := range j.localHistoryCache {
+			res, rsuccess := j.clientJudge.InspectHistory(v)
```

Comment:
does the struct passed into inspectHistory have a turn date, it might be beneficial for clients to know which turn history they are inpecting

Reply:
Good idea, added to InspectHistory function signature.